sdk/python/lib/pulumi/_utils.py

```diff
@@ -55,3 +58,31 @@ def is_empty_function(fn: typing.Callable) -> bool:
         (fn.__code__.co_code == _empty_lambda.__code__.co_code and consts == _consts_empty_lambda) or
         (fn.__code__.co_code == _empty_lambda_doc.__code__.co_code and consts == _consts_empty_lambda_doc)
     )
+
+
+# https://github.com/python/cpython/blob/master/Doc/library/importlib.rst#implementing-lazy-imports
+#
+# Original example extended to support import cycles and registration
+# of sub-modules as attributes.
+def _lazy_import(fullname):
```

Comment:
If we're going to be using this from our SDKs, I do wonder if we should not prefix it with `_` and possibly put it in a module that's not also prefixed with `_` to make it more clear that this called externally. `pulumi.runtime.lazy_import`?